presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcAggregatedPageSourceFactory.java

```diff
@@ -198,12 +198,14 @@ public static ConnectorPageSource createOrcPageSource(
             return new AggregatedOrcPageSource(physicalColumns, reader.getFooter(), typeManager, functionResolution);
```

Comment:
> Fix a resource leak for aggregated page source factories when no error is encountered. I think this leak might have been present before the refactoring too.

We need to either close the orcDataSource here, or pass it to the AggregatedOrcPageSource and let it close it in its `close` method.

Reply:
why isn't putting it in the finally block enough?

Reply:
Sorry, overlooked this. Closing in finally is a good idea.